packages/blade/src/components/Drawer/_decisions/decisions.md

```diff
@@ -0,0 +1,178 @@
+
+## Design
+
+- [Drawer - Figma Design](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade-DSL?node-id=78667%3A66663&mode=dev)
+
+## Proposed API
+
+```jsx
+<Drawer
+  isOpen={}
+  onDismiss={() => {}}
+  showOverlay
+>
+  <DrawerHeader
+    title=""
+    subtitle=""
+    leading={<DrawerHeaderIcon />}
+    titleSuffix={<DrawerHeaderBadge />}
+    trailing={
+      <>
+        <Link />
+        <Button />
+      </>
+    }
+  />
+  <DrawerBody>
+    <Slot />
+  </DrawerBody>
+<Drawer>
+```
+
+<details>
+<summary><b>Full Usage Example</b></summary>
+
+```jsx
+const MyCuteDrawer = () => {
+  const [showDrawer, setShowDrawer] = React.useState(false);
+  return (
+    <Box>
+      <Button onClick={() => setShowDrawer(true)}>Open Drawer</Button>
+      <Drawer
+        isOpen={showDrawer}
+        onDismiss={() => {
+          setShowDrawer(false);
+        }}
+      >
+        <DrawerHeader
+          title="Announcements"
+        />
+        <DrawerBody>
+          <FTXAnnouncement />
+          <RazorpayOnePromotions />
+          <CatPictures />
+        </DrawerBody>
+      <Drawer>
+    </Box>
+  )
+}
+
+```
+
+</details>
+
+### Props
+
+```ts
+type DrawerProps = {
+  /**
+   * Controlled state of drawer open or not
+   */
+  isOpen: boolean;
+
+  /**
+   * Dismiss handler
+   */
+  onDismiss: () => void;
+
+  /**
+   * Show or hide overlay.
+   *
+   * Also decides if clicking outside on overlay closes the drawer or not
+   */
+  showOverlay?: boolean;
+
+  /**
+   * Initial focus reference element
+   */
+  initialFocusRef?: React.MutableRefObject<any>;
+
+  /**
+   * children node.
+   *
+   * Supports DrawerHeader and DrawerBody
+   */
+  children: React.ReactNode;
+
+  /**
+   * Override z-index of Drawer.
+   *
+   * @default 1002
+   */
+  zIndex?: number;
+
+  /**
+   *  Accessibility label for the drawer
+   */
+  accessibilityLabel?: string;
+};
+
+type DrawerHeaderProps = {
+  /**
+   * Title of the Drawer
+   */
+  title: string;
+
+  /**
+   * Subtitle of the Drawer
+   */
+  subtitle?: string;
+
+  /**
+   * Leading element
+   *
+   * DrawerHeaderIcon or DrawerHeaderAsset
+   */
+  leading?: ReactNode;
+
+  /**
+   * Title suffix element
+   *
+   * DrawerHeaderBadge
+   */
+  titleSuffix?: ReactNode;
+
+  /**
+   * Title trailing element
+   *
+   * Link, Button[]
+   */
+  trailing?: ReactNode;
+};
+```
+
+Other supporting wrapper components for trailing and leading space-
+
+- DrawerHeaderBadge
+- DrawerHeaderIcon
+- DrawerHeaderAsset
+
+_No alternate APIs were considered because Drawer is closer to Modal on overall meaning and API perspective so made sense to go with API that is closer to Modal, also all DS Drawer components I referenced have similar API_
+
+## Drawer Stacking
+
+- Only 2 Drawers can be stacked on top of each other
+- 2nd Drawer always has overlay independent of `showOverlay` prop
+- 2nd Drawer always has back button instead of close button. Clicking on back button closes the 2nd drawer.
```

Comment:
Why though? a back button only makes sense when the navigation is _within_ the drawer and not another drawer on top of it.

If we are doing this, then we should also have a back button on 2nd Modal? Same logic would be applicable there. I think we shouldn't have a back button in either cases.

Reply:
In case of Modal, the size of the modal would also let people know about modal behind it no? If we put close button on 2nd Drawer, because all Drawers are same size and similarly placed, it feels like clicking close will close everything. But instead we only go back to the first Drawer in the stack.

Basically we would need something to let consumer know that this Drawer is stacked. Back button and overlay was our way to communicate that. But we can discuss this once with RK maybe?

Reply:
Started discussion here - https://razorpay.slack.com/archives/G01B3LQ9H0W/p1707813910309619?thread_ts=1707803235.144969&cid=G01B3LQ9H0W

Adding it to open questions